File: src/main/java/util/DateOperations.java
package util;

import java.text.*;
import java.util.Calendar;
import java.util.Locale;
import java.util.TimeZone;

public class DateOperations {

    public String getTodaysDateString() {
        Calendar today = getTodaysDate();
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
        return dateFormat.format(today.getTime());
    }

    public String getTuesdayAdjustedDateStringForDay(int dayOfWeekInt) {
        Calendar today = getTodaysDate();
        int dayOfWeekToday = today.get(Calendar.DAY_OF_WEEK);
        int dayOfWeekDiff = adjustToTuesdayStart(dayOfWeekInt) - adjustToTuesdayStart(dayOfWeekToday);
        today.add(Calendar.DATE, dayOfWeekDiff);
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
        return dateFormat.format(today.getTime());
    }

    public Calendar getTodaysDate() {
        return Calendar.getInstance();
    }

    private int adjustToTuesdayStart(int dayOfWeek) {
        return dayOfWeek > 2 ? dayOfWeek - 2 : dayOfWeek + 5;
    }

    public String getEasternTime(String dateString, String timeZone) {
        Calendar cal = Calendar.getInstance();
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH);
        sdf.setTimeZone(TimeZone.getTimeZone(timeZone));
        try {
            cal.setTime(sdf.parse(dateString));
        } catch (ParseException e) {
            e.printStackTrace();
        }
        DateFormat format = new SimpleDateFormat("EEE MM/dd h:mma z", Locale.US);
        format.setTimeZone(TimeZone.getTimeZone("EST"));
        return format.format(cal.getTime());
    }
}

File: src/main/kotlin/util/DateOperations.kt
package util

import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

fun getDateTodayString(): String {
    val today = getDateToday()
    val dateFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd")
    return dateFormat.format(today.time)
}

fun getDateToday(): Calendar {
    return Calendar.getInstance()
}

fun getEasternTime(dateString: String, timeZone: String, timePattern: String): String {
    val cal = Calendar.getInstance()
    val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH)
    sdf.timeZone = TimeZone.getTimeZone(timeZone)
    cal.time = sdf.parse(dateString)
    val format: DateFormat = SimpleDateFormat(timePattern, Locale.US)
    format.timeZone = TimeZone.getTimeZone("EST")
    return format.format(cal.time)
}